src/main/java/github/scarsz/discordsrv/linking/impl/system/sql/SqlAccountSystem.java

```diff
@@ -0,0 +1,420 @@
+                            ");")) {
+                statement.executeUpdate();
+            }
+        }
+    }
+
+    public abstract Connection getConnection();
+    public abstract boolean canStoreNativeUuids();
+
+    public void createDatabaseIfNotExists(String database) throws SQLException {
+        try (final PreparedStatement statement = getConnection().prepareStatement("CREATE DATABASE IF NOT EXISTS `" + database + "`")) {
+            statement.executeUpdate();
+        }
+    }
+
+    public boolean checkIfTableExists(String table) {
+        boolean tableExists = false;
+        try (final PreparedStatement statement = getConnection().prepareStatement("SELECT 1 FROM " + table + " LIMIT 1")) {
+            statement.executeQuery();
+            tableExists = true;
+        } catch (SQLException e) {
+            if (!e.getMessage().contains("doesn't exist")) DiscordSRV.error(e);
+        }
+        return tableExists;
+    }
+
+    public Map<String, String> getTableColumns(String table) throws SQLException {
+        final Map<String, String> columns = new HashMap<>();
+        try (final PreparedStatement statement = getConnection().prepareStatement("SHOW COLUMNS FROM " + table)) {
+            ResultSet result = statement.executeQuery();
+
+            while (result.next()) {
+                columns.put(result.getString("Field"), result.getString("Type"));
+            }
+        }
+        return columns;
+    }
+
+    public boolean checkIfTableMatchesStructure(String table, Map<String, String> expectedColumns) throws SQLException {
+        return checkIfTableMatchesStructure(table, expectedColumns, true);
+    }
+
+    public boolean checkIfTableMatchesStructure(String table, Map<String, String> expectedColumns, boolean showErrors) throws SQLException {
+        final List<String> found = new LinkedList<>();
+        for (Map.Entry<String, String> entry : getTableColumns(table).entrySet()) {
+            if (!expectedColumns.containsKey(entry.getKey())) continue; // only check columns that we're expecting
+            final String expectedType = expectedColumns.get(entry.getKey());
+            final String actualType = entry.getValue();
+            if (!expectedType.equals(actualType)) {
+                if (showErrors) {
+                    DiscordSRV.error("Expected type " + expectedType + " for column " + entry.getKey() + ", got " + actualType);
+                }
+                return false;
+            }
+            found.add(entry.getKey());
+        }
+
+        return found.containsAll(expectedColumns.keySet());
+    }
+
+    @Override
+    public void close() {
+        Connection connection = getConnection();
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    @SneakyThrows
```

Comment:
A lot of SneakyThrows on user facing methods (SQL- and RedisAccountSystems), I don't think it's a great idea to throw a SQLException to the end user here. Wrapping to a exception (checked or unchecked) that says that "DiscordSRV failed to lookup the linked user" or something along those lines would be nice.